app/index.tsx

```diff
@@ -1,67 +1,91 @@
 import { StatusBar } from 'expo-status-bar';
-import { Keyboard, ScrollView, StyleSheet, View } from 'react-native';
+import { Keyboard, StyleSheet, View, Text, ScrollView } from 'react-native';
 import NavButton from './components/NavButton';
 import PageHeader from './components/Header';
 import LabeledTextInput from './components/LabeledTextInput';
-import { getMatchData, updateMatchNumber, updateName, updateTeamNumber, updateDriverStation } from './api/data';
-import { DRIVER_STATION } from './api/data_types';
-import React, { useEffect, useState } from 'react';
+import { getMatchData, updateMatchNumber, updateName, updateTeamNumber, updateDriverStation, addUnsyncedData } from './api/data';
+import { DRIVER_STATION, MatchData } from './api/data_types';
+import { useEffect, useState } from 'react';
 import RadioButton from './components/RadioButton';
+import { onValue, ref, set } from 'firebase/database';
+import { db } from '../firebaseConfig';
+import AsyncStorage from '@react-native-async-storage/async-storage';
+import * as FileSystem from "expo-file-system";
 
 export default function App() {
     const [ nameFilled, setNameFilled ] = useState(false);
     const [ teamNumberFilled, setTeamNumberFilled ] = useState(false);
     const [ matchFilled, setMatchFilled ] = useState(false);
+    const [ eventCode, setEventCode ] = useState("");
 
     useEffect(() => {
-        const initializeRequirements = () => {
-            getMatchData().then((data) => {
-                setNameFilled(data["scouterName"] !== "");
-                setTeamNumberFilled(data["teamNumber"] !== 0);
-                setMatchFilled(data["matchNumber"] !== 0);
-            })
-        }
+        getMatchData().then((data) => {
+            setNameFilled(data["scouterName"] !== "");
+            setTeamNumberFilled(data["teamNumber"] !== 0);
+            setMatchFilled(data["matchNumber"] !== 0);
+        })
 
-        initializeRequirements();
-    })
+        onValue(ref(db, ".info/connected"), (snap) => {
+            if (snap.val() === false) return;
+
+            onValue(ref(db, "eventCode"), (code) => {
+                setEventCode(code.val());
+    
+                // this is nested in here to ensure that the eventCode is properly set, there may be a nicer way to do this that i dont know
+                AsyncStorage.getItem("unsynced").then((res) => {
+                    if (res === null) return; 
+    
+                    const unsyncedData = JSON.parse(res) as MatchData[];
+                    AsyncStorage.removeItem("unsynced");
+        
+                    unsyncedData.forEach((data) => {
+                        const path = `${code.val()}/${data["teamNumber"]}/${data["matchNumber"]}/${data["scouterName"]}`;
+                        set(ref(db, path), data);
```

Comment:
I think instead of doing a `set` in a loop you can do an `update`, but not sure. Will make the logic simpler